pages/api/authenticate/login.ts

```diff
@@ -26,41 +26,85 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
-        // password matched - create new token
-        const token = auth.tokenize(password);
+  // if user is not found, return early
+  if (!user) {
+    res.send(invalid);
+    res.end();
+    return false;
+  }
 
-        // set token in redis
-        client.setToken(token, {user: username, userID: user.id});
+  // if user found, verify password matches
+  if (auth.verifyPassword(password, user.password)) {
+    if (user.confirmation === "active") {
+      // password matched - create new token
+      const token = auth.tokenize(password);
 
-        // set HttpOnly cookies
-        res.setHeader('Set-Cookie', [`portal-token=${token}; path='/'; HttpOnly`, `portal-user=${username}; path='/'; HttpOnly`, `portal-user-id=${user.id}; path='/'; HttpOnly`]);
-        res.send(valid);
-        res.end();
-        return true;
-      } else {
-        res.send(unconfirmed);
+      // set token in redis
+      client.setToken(token, {user: username, userID: user.id});
+
+      // update last login with true flag
+      const updateLoginResp = await db.updateUser(user.id, {}, true);
+
+      if (!updateLoginResp.status) {
+        res.send(invalid);
         res.end();
-        return false;
       }
+
+      // set HttpOnly cookies
+      res.setHeader('Set-Cookie', [`portal-token=${token}; path='/'; HttpOnly`, `portal-user=${username}; path='/'; HttpOnly`, `portal-user-id=${user.id}; path='/'; HttpOnly`]);
+      res.send(valid);
+      res.end();
+      return true;
     } else {
-      // let user know this was an invalid request
-      res.send(invalid);
+      res.send(unconfirmed);
       res.end();
       return false;
     }
-  });
+  } else {
+    // let user know this was an invalid request
+    res.send(invalid);
+    res.end();
+    return false;
+  }
+
+  // db.getUser(escape(username)).then((user: any) => {
+  //   res.status(200);
+
+  //   // if user is not found, return early
+  //   if (!user) {
+  //     res.send(invalid);
+  //     res.end();
+  //     return false;
+  //   }
+
+  //   // if user found, verify password matches
+  //   if (auth.verifyPassword(password, user.password)) {
+  //     if (user.confirmation === "active") {
+  //       // password matched - create new token
+  //       const token = auth.tokenize(password);
+
+  //       // set token in redis
+  //       client.setToken(token, {user: username, userID: user.id});
+
+  //       // set HttpOnly cookies
+  //       res.setHeader('Set-Cookie', [`portal-token=${token}; path='/'; HttpOnly`, `portal-user=${username}; path='/'; HttpOnly`, `portal-user-id=${user.id}; path='/'; HttpOnly`]);
+  //       res.send(valid);
+  //       res.end();
+  //       return true;
+  //     } else {
+  //       res.send(unconfirmed);
+  //       res.end();
+  //       return false;
+  //     }
+  //   } else {
+  //     // let user know this was an invalid request
+  //     res.send(invalid);
+  //     res.end();
+  //     return false;
+  //   }
+  // });
```

Comment:
@JacobSNGoodwin was about to merge into dev, but noticed this was commented out. Was this for debugging or? Was this replaced with something else? Thanks!

Reply:
@Andymendez100... cuz of I is bozo.

@lloan oops. Yes, I actually changed it to async/await because it was getting very messy, so I kept that there while editing. Don't know how I missed it. 

I could have returned a value and passed it to the next then block, but it was starting to be hard to read because I update the time stamp basically in the middle of the flow. 

I just removed the comments.